include/oneapi/dpl/pstl/parallel_backend.h

```diff
@@ -15,42 +15,41 @@
 #ifndef _ONEDPL_PARALLEL_BACKEND_H
 #define _ONEDPL_PARALLEL_BACKEND_H
 #include "onedpl_config.h"
+
+// Select a parallel backend
+#if ONEDPL_USE_TBB_BACKEND || (!defined(ONEDPL_USE_TBB_BACKEND) && !ONEDPL_USE_OPENMP_BACKEND && _ONEDPL_TBB_AVAILABLE)
+#    define _ONEDPL_PAR_BACKEND_TBB 1
+#    include "parallel_backend_tbb.h"
+#elif ONEDPL_USE_OPENMP_BACKEND || (!defined(ONEDPL_USE_OPENMP_BACKEND) && _ONEDPL_OPENMP_AVAILABLE)
+#    include "parallel_backend_omp.h"
+#    define _ONEDPL_PAR_BACKEND_OPENMP 1
+#else
+#    include "parallel_backend_serial.h"
+#    define _ONEDPL_PAR_BACKEND_SERIAL 1
+#endif
+
 #if _ONEDPL_BACKEND_SYCL
 #    include "hetero/dpcpp/parallel_backend_sycl.h"
 #    if _ONEDPL_FPGA_DEVICE
 #        include "hetero/dpcpp/parallel_backend_sycl_fpga.h"
 #    endif
 #endif
 
-#if _ONEDPL_PAR_BACKEND_TBB
-#    include "parallel_backend_tbb.h"
 namespace oneapi
 {
 namespace dpl
 {
-namespace __par_backend = __tbb_backend;
-}
-} // namespace oneapi
+namespace __par_backend =
+#if _ONEDPL_PAR_BACKEND_TBB
+    __tbb_backend;
 #elif _ONEDPL_PAR_BACKEND_OPENMP
-#    include "parallel_backend_omp.h"
-namespace oneapi
-{
-namespace dpl
-{
-namespace __par_backend = __omp_backend;
-}
-} // namespace oneapi
+    __omp_backend;
 #elif _ONEDPL_PAR_BACKEND_SERIAL
-#    include "parallel_backend_serial.h"
-namespace oneapi
-{
-namespace dpl
-{
-namespace __par_backend = __serial_backend;
-}
-} // namespace oneapi
+    __serial_backend;
 #else
-_PSTL_PRAGMA_MESSAGE("Parallel backend was not specified");
+    _PSTL_PRAGMA_MESSAGE("Parallel backend was not specified");
 #endif
```

Comment:
The `#else` case is effectively unreachable code, as the backend selection logic at lines 19-29 ensures that exactly one of the `_ONEDPL_PAR_BACKEND_*` macros is defined.

Reply:
Yes, and in line 50 we are checking that some of `_ONEDPL_PAR_BACKEND_*` has been defined.

Reply:
The line 50 checks nothing, it's unreachable code. Moreover, it's an incorrect code as far as I can tell, because the "result" of a pragma cannot be used to set the par_backend namespace alias at line 42.

Reply:
Ok, with `#error` it makes a little more sense, at least the intent is clearly visible.